simtools/simtel/simtel_histograms.py

```diff
@@ -1,26 +1,548 @@
 import copy
 import logging
+from pathlib import Path
 
 import numpy as np
 from astropy import units as u
+from astropy.table import QTable
+from ctao_cr_spectra.definitions import IRFDOC_PROTON_SPECTRUM
+from ctao_cr_spectra.spectral import cone_solid_angle
 from ctapipe.io import write_table
 from eventio import EventIOFile, Histograms
 from eventio.search_utils import yield_toplevel_of_type
+from eventio.simtel import MCRunHeader
 
 from simtools import version
 from simtools.io_operations.hdf5_handler import fill_hdf5_table
 from simtools.utils.names import sanitize_name
 
-__all__ = ["InconsistentHistogramFormat", "SimtelHistograms"]
+__all__ = [
+    "InconsistentHistogramFormat",
+    "HistogramIdNotFound",
+    "SimtelHistogram",
+    "SimtelHistograms",
+]
 
 
 class InconsistentHistogramFormat(Exception):
     """Exception for bad histogram format."""
 
 
+class HistogramIdNotFound(Exception):
+    """Exception for histogram ID not found."""
+
+
+class SimtelHistogram:
+    """
+    This class handles a single histogram (or simtel_array output) file.
+    """
+
+    trigger_rate: None
+
+    def __init__(self, histogram_file, rht=False):
+        """
+        Initialize SimtelHistogram class.
+        """
+        self._logger = logging.getLogger(__name__)
+        self.histogram_file = histogram_file
+        if not Path(histogram_file).exists():
+            msg = f"File {histogram_file} does not exist."
+            self._logger.error(msg)
+            raise FileNotFoundError
+        self._config = None
+        self._view_cone = None
+        self._total_area = None
+        self._solid_angle = None
+        self._energy_range = None
+        self._total_num_simulated_events = None
+        self._total_num_triggered_events = None
+        self._histogram = None
+        self._histogram_file = None
+        self._initialize_histogram()
+        self.trigger_rate = None
+        self.trigger_rate_uncertainty = None
+        self.trigger_rate_per_energy_bin = None
+        self.energy_axis = None
+        self.radius_axis = None
+        self.rht = rht  # If true, area thrown is estimated exactly as in the hessio rht.cc tool.
+
+    def _initialize_histogram(self):
+        """
+        Initializes lists of histograms and files.
+
+        Returns
+        -------
+        list:
+            List of histograms.
+        """
+        with EventIOFile(self.histogram_file) as f:
+            for obj in yield_toplevel_of_type(f, Histograms):
+                self.histogram = obj.parse()
+
+    @property
```

Comment:
As discussed, yes, they are useful as these are intrinsic properties of the class, which make them slightly different than a simple class attribute or variable which we can change as we see fit. Also, the use of properties is not conditioned to the use of setters. The property, e.g. number of simulated events, is intrinsic to the simtel_array file (should not be changed). Therefore, the num of simulated events is better used as a property rather than attribute. If you want to know more I suggest you look into more details in this regard to get more into this "pythonic way": https://www.linkedin.com/pulse/understanding-property-decorator-python-nuno-bispo-m7aee/

Reply:
Where do you use the functionality provided by @property? I didn't see that.

Reply:
I don't think you checked the text I sent above, but if you check it you will see that, for example, the property provides protection against being changed, what an attribute does not have. Trying to change e.g. the value of `num_simulated_events` will not work, as it should not. I do not want to spend more time in this discussion, the use of property is well accepted, motivated and supported throughout the python community. So unless you have an argument against it (e.g. slower code?), please let us go ahead with this.

Reply:
Let's not waste time on it anymore - I do want to have code as simple as possible with only necessary feature. Experience shows that this is the key to long-term maintainability. If we would worry about attribute protection, we wouldn't choose python but a language which would provide this natively. Anyway, let's stop here.